src/perturb.jl

```diff
@@ -78,17 +78,17 @@ function create_perturbation_gates(s, B_pert, α, x, L, N_sites, energy_sign, τ
 end
 
 
-function evolve_perturbation(s, τ, B_pert, α, x, L, N_sites, ψ, cutoff, maxdim, energy_sign,ttotal)
+function evolve_perturbation(s,k, τ_pert, B_pert, α, x, L, N_sites, ψ, cutoff, maxdim, energy_sign,ttotal)
```

Comment:
There shouldn't be a separate evolve function, I think. You should apply the perturbation in a single step. w and tau shouldn't be separate arguments - you should pass a single parameter called "amplitude", which is equivalent to (w*tau)